src/op.rs

```diff
@@ -0,0 +1,2691 @@
+            assert_eq!(valbytes.len(), bytecode.len());
+        }
+        otherwise => panic!("Unexpected result: {:?}", otherwise),
+    }
+
+    match compute_pc(ebuf, bytecode, -1) {
+        Err(Error::BadBranchTarget(_)) => {}
+        otherwise => panic!("Unexpected result: {:?}", otherwise),
+    }
+
+    match compute_pc(ebuf, bytecode, 5) {
+        Ok(val) => {
+            let valbytes: &[u8] = val.into();
+            assert_eq!(valbytes.len(), 0);
+        }
+        otherwise => panic!("Unexpected result: {:?}", otherwise),
+    }
+
+    match compute_pc(EndianBuf::<LittleEndian>::new(&bytes[3..]), bytecode, -2) {
+        Ok(val) => {
+            let valbytes: &[u8] = val.into();
+            assert_eq!(valbytes.len(), bytecode.len() - 1);
+        }
+        otherwise => panic!("Unexpected result: {:?}", otherwise),
+    }
+
+    match compute_pc(EndianBuf::<LittleEndian>::new(&bytes[2..]), bytecode, 2) {
+        Ok(val) => {
+            let valbytes: &[u8] = val.into();
+            assert_eq!(valbytes.len(), bytecode.len() - 4);
+        }
+        otherwise => panic!("Unexpected result: {:?}", otherwise),
+    }
+}
+
+// Like parse_u8 but takes and returns an EndianBuf for convenience.
+fn parse_u8e<'input, Endian>(bytes: EndianBuf<'input, Endian>)
+                             -> ParseResult<(EndianBuf<'input, Endian>, u8)>
+    where Endian: Endianity
+{
+    let (bytes, value) = try!(parse_u8(bytes.into()));
+    Ok((EndianBuf::new(bytes), value))
+}
+
+// Like parse_i8 but takes and returns an EndianBuf for convenience.
+fn parse_i8e<'input, Endian>(bytes: EndianBuf<'input, Endian>)
+                             -> ParseResult<(EndianBuf<'input, Endian>, i8)>
+    where Endian: Endianity
+{
+    let (bytes, value) = try!(parse_i8(bytes.into()));
+    Ok((EndianBuf::new(bytes), value))
+}
+
+// Like parse_unsigned_leb but takes and returns an EndianBuf for
+// convenience.
+fn parse_unsigned_lebe<'input, Endian>(bytes: EndianBuf<'input, Endian>)
+                                       -> ParseResult<(EndianBuf<'input, Endian>, u64)>
+    where Endian: Endianity
+{
+    let (bytes, value) = try!(parse_unsigned_leb(bytes.into()));
+    Ok((EndianBuf::new(bytes), value))
+}
+
+// Like parse_signed_leb but takes and returns an EndianBuf for
+// convenience.
+fn parse_signed_lebe<'input, Endian>(bytes: EndianBuf<'input, Endian>)
+                                     -> ParseResult<(EndianBuf<'input, Endian>, i64)>
+    where Endian: Endianity
+{
+    let (bytes, value) = try!(parse_signed_leb(bytes.into()));
+    Ok((EndianBuf::new(bytes), value))
+}
+
+/// Decode a single DWARF expression operation.
+///
+/// This is useful when examining a DWARF expression for reasons other
+/// than direct evaluation.
+///
+/// `bytes` points to a the operation to decode.  It should point into
+/// the same array as `bytecode`, which should be the entire
+/// expression.
+pub fn decode_expression_op<'input, Endian>
```

Comment:
I prefer the parse functions to be static members of the struct that they parse. eg `Operation::parse`. I don't feel strongly about that, but we should be consistent.